Add ML commons settings (#2063)
opensearch-project/documentation-website: _ml-commons-plugin/cluster-settings.md

nav_order: 10
---

# ML Commons cluster settings

This page provides an overview of `opensearch.yml` settings that can be configured for the ML commons plugin.

## Run tasks and models on ML nodes only

### Setting

```
plugins.ml_commons.only_run_on_ml_node: false
```

### Description

If `true`, ML Commons tasks and models run machine learning (ML) tasks on ML nodes only. If `false`, tasks and models run on ML nodes first. If no ML nodes exist, tasks and models run on data nodes. Don't set as "false" on production cluster. 

### Values

- Default value: `false`
- Value range: `true` or `false`

## Dispatch tasks to ML node with least load

### Setting

```
plugins.ml_commons.task_dispatch_policy: round_robin
```

### Description

`round_robin` dispatches ML tasks to ML nodes using round robin routing. `least_load` gathers all an ML nodes' runtime information, like JVM heap memory usage and running tasks, then dispatches tasks to the ML node with the least load. 

### Values

- Dafault value: `round_robin`
- Value range: `round_robin or least_load`

## Set sync up job intervals 

### Setting

```
plugins.ml_commons.sync_up_job_interval_in_seconds: 10
```

### Description

> When returning runtime information with the [profile API]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api#profile), ML Commons will run a regular sync up job when models are added or removed on each node. When set to `0`, ML Commons immediately stops sync up jobs.

Reviewer: ```suggestion
When returning runtime information with the [profile API]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api#profile), ML Commons will run a regular sync up job to sync up newly loaded/unloaded models on each node. When set to `0`, ML Commons immediately stops sync up jobs.
```
